scipy/stats/_multivariate.py

```diff
@@ -5067,3 +5068,257 @@ def rvs(self, size=1, random_state=None):
         method.__doc__, mhg_docdict_noparams)
     method.__doc__ = doccer.docformat(method.__doc__,
                                       mhg_docdict_params)
+
+
+class conditional_table_gen(multi_rv_generic):
+    r"""Distribution of random tables with given marginals.
+
+    This is the distribution of random tables with given row and column vector
+    sums. This distribution represents the set of random tables under the null
+    hypothesis that rows and columns are independent. It is used in hypothesis
+    tests of independence.
+
+    Because of assumed independence, the expected frequency of each table
+    element can be computed from the row and column sums, so that the
+    distribution is completely determined by these two vectors.
+
+    Methods
+    -------
+    mean(row, col)
+        Mean table.
+    rvs(row, col, size=1, random_state=None)
```

Comment:
This is missing `method`. I'd either include it here and test the behavior (including `NotImplementedError`) or remove it until it's really implemented.

Reply:
Fixed. Since we are discussing the API, I think we should have `method` from the start. I forgot to add it here.

Reply:
For me personally, it is essential that we add patefield's algorithm, since the contingency tables aka histograms that we use in high-energy physics typically have entries in the millions.

Reply:
Sure. I'm glad we split this up, though. Even without algorithmic code, this is already at 500 lines!